### ISSUE TITLE
Kali 2.0 - 

### ISSUE THREAD

**Issue by frafrichile** [aanarchyy/wifite-mod-pixiewps#31]
Hi, 
Great tool . I have used it plenty of times on Kali 1.0

But on Kali 2.0 it just says

enabling monitor mode on phy1
enabling monitor mode on phy1
enabling monitor mode on phy1

any ideas?

Thank you !

**Comment by aanarchyy** [comment]
I will have to download Kali 2.0 and test this.  Will address shortly
Make sure you are using the newest revision of wifite-ng
That sounds like the problem with the new aircrack-ng that was fixed in  [<61f97e7>](https://github.com/aanarchyy/wifite-mod-pixiewps/commit/2d9edd8a4706a7233c588bfcc9350d532e884472)

**Comment by postak05** [comment]
I can confirm this problem. Solution is:
1] run "original" wifite, which is included in Kali2.0 by default
2] wait few seconds till monitor mode is activated
3] break "original" wifite
4] run wifite-ng

it seems to me, that "original" wifite (r87) somehow prepare WiFi card. I had similar problem when I tried to start monitor mode using airmon-ng. No monitor mode started, wifite fixed that problem for me every session, so now I always run "original" wifite at first after boot Kali 2.0.

**Comment by frafrichile** [comment]
Hi thanks for checking that.
Another solution is to manually start the monitor interface. 

I was wondering  is it possible to just run pixiewps attack. Maybe an enhancement in the future. 

Thanks!

**Comment by postak05** [comment]
I've got it!! (I've looked into "original" wifite)
look at wifite-ng (ver=112), function "enable_monitor_mode(iface)", line 957:

call(['airmon-ng', 'start', iface], stdout=DN, stderr=DN)

change it to (add 1 line in front):

call(['airmon-ng', 'check', 'kill'], stdout=DN, stderr=DN)
call(['airmon-ng', 'start', iface], stdout=DN, stderr=DN)

**Comment by FractumSeraph** [comment]
I'm on Ubuntu 15.04, and whenever I use that method I lose internet access after I run Wifite.  This is because it's killing my network manager.   Make sure to also have it restart the NetworkManager and wpa_supplicant services as soon as it's done.   I tried doing this myself but it didn't seem to work.  I added this after the "Disabling monitor mode" dialog:

print 'Restarting NetworkManager and wpa_supplicant'
    call(['service', 'NetworkManager', 'start'], stdout=DN, stderr=DN)
    call(['service', 'wpa_supplicant', 'start'], stdout=DN, stderr=DN)

I don't know the language, so maybe I was just doing something wrong.

And even better way would be to replace the check, then kill, with something that stored what was currently running into an array, then restart all of the services in the array after wifite was done.

TL;DR
If you do this you will lose internet after running Wifite.  To fix it, run these (As root/sudo):
service NetworkManager restart
service wpa_supplicant restart

**Comment by 555STi** [comment]
+1 for @FallenArchangel, same issue here, each time I stop using WiFite need to restart the network-manager service.

I have Kali 2.0 with WiFite v2 (r87).
